tests/nvmeof/test_ceph_nvmeof_high_availability.py

```diff
@@ -0,0 +1,313 @@
+            _io_args.update(
+                {
+                    "device_name": target["DevicePath"],
+                    "client_node": client,
+                    "long_running": True,
+                    "cmd_timeout": "notimeout",
+                }
+            )
+            _io_args = {**io_args, **_io_args}
+            p.spawn(run_fio, **_io_args)
+        for op in p:
+            results.append(op)
+    return results
+
+
+def disconnect_initiator(ceph_cluster, node, subnqn):
+    """Disconnect Initiator."""
+    node = get_node_by_id(ceph_cluster, node)
+    initiator = Initiator(node)
+    initiator.disconnect(**{"nqn": subnqn})
+
+
+def teardown(ceph_cluster, rbd_obj, nvmegwcli, config):
+    """Cleanup the ceph-nvme gw entities.
+
+    Args:
+        ceph_cluster: Ceph Cluster
+        rbd_obj: RBD object
+        config: test config
+    """
+    # Delete the multiple Initiators across multiple gateways
+    if "initiators" in config["cleanup"]:
+        for initiator_cfg in config["initiators"]:
+            disconnect_initiator(
+                ceph_cluster, initiator_cfg["node"], initiator_cfg["subnqn"]
+            )
+
+    # # Delete the multiple subsystems across multiple gateways
+    # if "subsystems" in config["cleanup"]:
+    #     config_sub_node = config["subsystems"]
+    #     if not isinstance(config_sub_node, list):
+    #         config_sub_node = [config_sub_node]
+    #     for sub_cfg in config_sub_node:
+    #         node = config["gw_node"] if "node" not in sub_cfg else sub_cfg["node"]
+    #         LOG.info(f"Deleting subsystem {sub_cfg['nqn']} on gateway {node}")
+    #         nvmegwcli.subsystem.delete(
+    #             **{"args": {"subsystem": sub_cfg["nqn"], "force": True}}
+    #         )
+
+    # Delete the gateway
+    if "gateway" in config["cleanup"]:
+        cfg = {
+            "no_cluster_state": False,
+            "config": {
+                "command": "remove",
+                "service": "nvmeof",
+                "args": {"service_name": f"nvmeof.{config['rbd_pool']}"},
+            },
+        }
+        test_orch.run(ceph_cluster, **cfg)
+
+    # Delete the pool
+    if "pool" in config["cleanup"]:
+        rbd_obj.clean_up(pools=[config["rbd_pool"]])
+
+
+def run(ceph_cluster: Ceph, **kwargs) -> int:
+    """Return the status of the Ceph NVMEof HA test execution.
+
+    - Configure SPDK and install with control interface.
+    - Configures Initiators and Run FIO on NVMe targets.
+
+    Args:
+        ceph_cluster: Ceph cluster object
+        kwargs: Key/value pairs of configuration information to be used in the test.
+
+    Returns:
+        int - 0 when the execution is successful else 1 (for failure).
+
+    Example:
+
+        # Execute the nvmeof GW test
+            - test:
+                name: Ceph NVMeoF deployment
+                desc: Configure NVMEoF gateways and initiators
+                config:
+                    gw_node: node6
```

Comment:
Just a small doc correction here. You have used gw_nodes as the name to parse and also in test suite. So could you please change gw_node to gw_nodes here as well?

Reply:
updated